### DISCUSSION TITLE
How to define function index in sequelize model?

### DISCUSSION THREAD

**Discussion by mjy78** [sequelize/sequelize#15243] [Help & Questions]
I need to define a functional index against a spatial field in postgres. The index statement looks like this...

```
CREATE INDEX IF NOT EXISTS coverage_segments_path
    ON public.coverage_segments USING gist
    (ST_Transform(ST_SetSRID(path, 4326), 3857))
    TABLESPACE pg_default;
	
```

However if I define the index in the model as such...

```
   indexes: [
      {
        fields: ['ST_Transform(ST_SetSRID(path, 4326), 3857)'],
        using: 'GIST'
      }
```

I get an error that "column ST_Transform(ST_SetSRID(path, 4326), 3857) does not exist" 

Is there a way to define functional indexes in a Sequelize model definition?

**Comment by ephys** [comment]
I assume sequelize produces the following query (you use `logging: true` on your `sequelize.sync()` call to check):

```sql
CREATE INDEX IF NOT EXISTS coverage_segments_path
    ON public.coverage_segments USING gist
    ("ST_Transform(ST_SetSRID(path, 4326), 3857)")
    TABLESPACE pg_default;
```

It's possible you could bypass this quoting by using a literal:

```ts
import { literal } from 'sequelize';

// ...

indexes: [{
  fields: [literal('ST_Transform(ST_SetSRID(path, 4326), 3857)')],
  using: 'GIST'
}]
```

If that causes a crash due to `literal` not being expected, we can open a feature request to support it
